presto-mongodb/src/test/java/io/prestosql/plugin/mongodb/TestMongoIntegrationSmokeTest.java

```diff
@@ -357,6 +359,19 @@ public void testNullPredicates()
         assertUpdate("DROP TABLE test.null_predicates");
     }
 
+    @Test
+    public void testLimitPushdown()
+    {
+        assertThat(query("SELECT name FROM nation LIMIT 30")).isFullyPushedDown(); // Use high limit for result determinism
```

Comment:
@ebyhr Hi, do we support SQL with both where condition and limit, e.g. "SELECT name FROM nation where nation.aa='xxx' LIMIT 30"?
I am testing using 424 version, and it does not work like this

Reply:
It works as far as I confirmed. Could you share the table definition and EXPLAN result?

Reply:
Thanks a lot for your quick response.
The content in the MongoDB table is pretty simple:
`
{
  "_id": {
    "$oid": "650168eb02778e01c2832e8b"
  },
  "information_package_iri": "https://domain/api/informationPackages/H5XW3DXTKNH5TOXM4DPU75VJY4",
  "content_identifier": "75fbfb39-9b5e-4fa3-801b-37745dc60b9e",
  "column0": "52",
  "column1": "96",
  "column10": "50",
  "column11": "15",
  "column12": "64",
  "column13": "62",
  "column14": "89",
  "column2": "17",
  "column3": "74",
  "column4": "9",
  "column5": "5",
  "column6": "65",
  "column7": "30",
  "column8": "94",
  "column9": "86",
  "iri": "urn:uuid:010b9f55-db02-4be5-9c01-c6ad4bc926c1"
}
`
And the following is the explain result:
`
Trino version: 424
Fragment 0 [SINGLE]
    Output layout: [column0, column1, column10, column11, column12, column13, column14, column2, column3, column4, column5, column6, column7, column8, column9, content_identifier, information_package_iri, iri]
    Output partitioning: SINGLE []
    Output[columnNames = [column0, column1, column10, column11, column12, column13, column14, column2, column3, column4, column5, column6, column7, column8, column9, content_identifier, information_package_iri, iri]]
    │   Layout: [column0:varchar, column1:varchar, column10:varchar, column11:varchar, column12:varchar, column13:varchar, column14:varchar, column2:varchar, column3:varchar, column4:varchar, column5:varchar, column6:varchar, column7:varchar, column8:varchar, column9:varchar, content_identifier:varchar, information_package_iri:varchar, iri:varchar]
    │   Estimates: {rows: 5 (4.83kB), cpu: 0, memory: 0B, network: 0B}
    └─ Limit[count = 5]
       │   Layout: [column0:varchar, column1:varchar, column10:varchar, column11:varchar, column12:varchar, column13:varchar, column14:varchar, column2:varchar, column3:varchar, column4:varchar, column5:varchar, column6:varchar, column7:varchar, column8:varchar, column9:varchar, content_identifier:varchar, information_package_iri:varchar, iri:varchar]
       │   Estimates: {rows: 5 (4.83kB), cpu: 4.83k, memory: 0B, network: 0B}
       └─ LocalExchange[partitioning = SINGLE]
          │   Layout: [column0:varchar, column1:varchar, column10:varchar, column11:varchar, column12:varchar, column13:varchar, column14:varchar, column2:varchar, column3:varchar, column4:varchar, column5:varchar, column6:varchar, column7:varchar, column8:varchar, column9:varchar, content_identifier:varchar, information_package_iri:varchar, iri:varchar]
          │   Estimates: {rows: 5 (4.83kB), cpu: 0, memory: 0B, network: 0B}
          └─ RemoteSource[sourceFragmentIds = [1]]
                 Layout: [column0:varchar, column1:varchar, column10:varchar, column11:varchar, column12:varchar, column13:varchar, column14:varchar, column2:varchar, column3:varchar, column4:varchar, column5:varchar, column6:varchar, column7:varchar, column8:varchar, column9:varchar, content_identifier:varchar, information_package_iri:varchar, iri:varchar]

Fragment 1 [SOURCE]
    Output layout: [column0, column1, column10, column11, column12, column13, column14, column2, column3, column4, column5, column6, column7, column8, column9, content_identifier, information_package_iri, iri]
    Output partitioning: SINGLE []
    LimitPartial[count = 5]
    │   Layout: [column0:varchar, column1:varchar, column10:varchar, column11:varchar, column12:varchar, column13:varchar, column14:varchar, column2:varchar, column3:varchar, column4:varchar, column5:varchar, column6:varchar, column7:varchar, column8:varchar, column9:varchar, content_identifier:varchar, information_package_iri:varchar, iri:varchar]
    │   Estimates: {rows: 5 (4.83kB), cpu: 4.83k, memory: 0B, network: 0B}
    └─ ScanFilter[table = data_description:stresstest.test_table stresstest.test_table Optional.empty  constraint on [information_package_iri] nextSyntheticColumnId=0, filterPredicate = ("information_package_iri" = VARCHAR 'https://domain/api/informationPackages/H5XW3DXTKNH5TOXM4DPU75VJY4')]
           Layout: [column0:varchar, column1:varchar, column10:varchar, column11:varchar, column12:varchar, column13:varchar, column14:varchar, column2:varchar, column3:varchar, column4:varchar, column5:varchar, column6:varchar, column7:varchar, column8:varchar, column9:varchar, content_identifier:varchar, information_package_iri:varchar, iri:varchar]
           Estimates: {rows: ? (?), cpu: ?, memory: 0B, network: 0B}/{rows: ? (?), cpu: ?, memory: 0B, network: 0B}
           column1 := column1:varchar
           column12 := column12:varchar
           column0 := column0:varchar
           column11 := column11:varchar
           iri := iri:varchar
           column10 := column10:varchar
           column5 := column5:varchar
           column4 := column4:varchar
           column3 := column3:varchar
           information_package_iri := information_package_iri:varchar
           column2 := column2:varchar
           column14 := column14:varchar
           column13 := column13:varchar
           content_identifier := content_identifier:varchar
           column9 := column9:varchar
           column8 := column8:varchar
           column7 := column7:varchar
           column6 := column6:varchar

`

Reply:
That's weird. Could you share the entire steps to reproduce? CREATE TABLE, INSERT & SELECT statements.